Source/WebCore/rendering/style/PositionArea.h

```diff
@@ -25,43 +25,76 @@
  */
 
 #pragma once
+#include "BoxSides.h"
+#include "WritingMode.h"
 
 namespace WebCore {
 
 // The axis that the span specifies.
+// Represented as 3 bits, with the last one indicating the corresponding [Logical]BoxAxis value.
 enum class PositionAreaAxis : uint8_t {
-    // Physical axes. Implies self is PositionAreaSelf::No, as physical axes
-    // do not depend on writing mode of any elements.
-    Horizontal,
-    Vertical,
-
-    // Logical axes.
-    X,
-    Y,
-    Block,
-    Inline,
-};
+    // Physical axes × Physical directions.
+    Horizontal = 0b000,
+    Vertical   = 0b001,
 
-// Specifies which tile(s) on the axis that the position-area span occupies.
-enum class PositionAreaTrack : uint8_t {
-    // First tile.
-    Start,
+    // Physical axes × Logical directions.
+    X = 0b010,
+    Y = 0b011,
 
-    // First and center tiles.
-    SpanStart,
+    // Logical axes × Logical directions.
+    Inline = 0b110,
+    Block  = 0b111,
+};
 
-    // Last tile.
-    End,
+static inline bool isPositionAreaAxisLogical(const PositionAreaAxis positionAxis)
+{
+    return static_cast<uint8_t>(positionAxis) & 0b100;
+}
+
+static inline bool isPositionAreaDirectionLogical(const PositionAreaAxis positionAxis)
+{
+    return static_cast<uint8_t>(positionAxis) & 0b010;
+}
+
+static inline BoxAxis mapPositionAreaAxisToPhysicalAxis(const PositionAreaAxis positionAxis, const WritingMode writingMode)
+{
+    auto physicalAxis = static_cast<uint8_t>(positionAxis) & 0b001;
```

Comment:
Maybe put this bitwise into a separate function (like `isPositionAreaDirectionLogical`)? So we'd have three functions for each bit, and it's a bit easier to read than a bitwise.

Reply:
I don't want anyone else (outside of these methods that "belong" to PositionAreaAxis) to be extracting out this bit. :) So I don't want to create a function. I can name it or add a comment though.

Reply:
Yeah, I don't love the duplication, but this is a .h file, so even a static method is exposed to anything that imports the enum, right?

Reply:
That's true, I neglect this is a header file. Let's keep thing the current way then.